Admin/Test-AMSI.ps1

```diff
@@ -1,144 +1,891 @@
                 $host.ui.RawUI.ForegroundColor = "Green"
-                Write-Output "---------------------------------------------------------------------------------------------------------------"
                 $host.ui.RawUI.ForegroundColor = "Yellow"
-                Write-Output "This may be indicative of a potential block from AMSI"
+                Write-Host "This may be indicative of a potential block from AMSI"
                 $host.ui.RawUI.ForegroundColor = "Green"
-                $msgCheckLogs = "Check your log files located in " + $installPath + "Logging\HttpRequestFiltering\"
-                Write-Output $msgCheckLogs
-                $msgDetectedTimeStamp = "for a Detected result around " + $dateTime.ToUniversalTime()
-                Write-Output $msgDetectedTimeStamp
+                if ($IsServer) {
+                    $getMSIInstallPathSB = { (Get-ItemProperty -Path Registry::HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\ExchangeServer\v15\Setup -ErrorAction SilentlyContinue).MsiInstallPath }
+                    $ExchangePath = Invoke-ScriptBlockHandler -ComputerName $ExchangeServer -ScriptBlock $getMSIInstallPathSB
+                    $msgCheckLogs = "You can check your log files located in $($ExchangePath)Logging\HttpRequestFiltering\"
+                } else {
+                    $msgCheckLogs = "You can check your log files located in %ExchangeInstallPath%\Logging\HttpRequestFiltering\ in all server included in $ExchangeServer endpoint"
+                }
+                Write-Host $msgCheckLogs
+                $host.ui.RawUI.ForegroundColor = $currentForegroundColor
+                $msgDetectedTimeStamp = "You should find result around $((Get-Date).ToUniversalTime().ToString("M/d/yyy h:mm:ss tt")) UTC"
+                Write-Host $msgDetectedTimeStamp
+                if ($IsServer) {
+                    Write-Host ""
+                    Write-Host "Checking logs on $server at $($testTime.ToString("M/d/yyy h:mm:ss tt"))"
+                    $HttpRequestFilteringLogFolder = $null
+
+                    if ($ExchangePath) {
+                        if ($server.Equals($env:COMPUTERNAME, [System.StringComparison]::OrdinalIgnoreCase)) {
+                            $HttpRequestFilteringLogFolder = Join-Path $ExchangePath "Logging\HttpRequestFiltering\"
+                        } else {
+                            $HttpRequestFilteringLogFolder = Join-Path "\\$server\$($ExchangePath.Replace(':','$'))" "Logging\HttpRequestFiltering\"
+                        }
+                    } else {
+                        Write-Host "Cannot get Exchange installation path on $server" -ForegroundColor Red
+                    }
+
+                    if (Test-Path $HttpRequestFilteringLogFolder -PathType Container) {
+                        $file = $null
+                        $timeout = (Get-Date).AddMinutes(1)
+                        $detected = $false
+                        $marginTime = New-TimeSpan -Seconds 5
```

Comment:
It might be easier to use a timestamp/ correlation id to locate the entry in the log. Something like this: 
![image](https://github.com/microsoft/CSS-Exchange/assets/40993616/f7c58f40-5552-461d-97ed-629e4520a1bd)

Reply:
If i use an id in the request it is easier to identify and do not need to include margin time.
That simplify the code a lot.